lib/ecto/repo/schema.ex

```diff
@@ -761,13 +761,23 @@ defmodule Ecto.Repo.Schema do
   end
 
   defp process_children(changeset, assocs, user_changeset, adapter, opts) do
+    opts = process_children_opts(opts)
+
     case Ecto.Association.on_repo_change(changeset, assocs, adapter, opts) do
       {:ok, struct} -> {:ok, struct}
       {:error, changes} ->
         {:error, %{user_changeset | valid?: false, changes: changes}}
     end
   end
 
+  defp process_children_opts(opts) do
+    if is_list(Keyword.get(opts, :returning)) do
```

Comment:
IMO it is fine to always drop it.

Reply:
Then we will disable all `read_after_write`s that are defined for assocs. If that's okay - I can rollback last commit.

Reply:
Oh. because the option can be a boolean? I see. Let's leave it as is then. Good catch. Let's just write the unit tests then, remove the integration ones, and we should be good to go. Ping me (here on IRC) if you have any issues with the tests. :)

Reply:
To be fair. that was @wojtekmach catch :)